### PR TITLE
Replaced moment js with day js

### PR DESCRIPTION
**Pull request by Julias0**

### CODE REVIEW

File: src/app/core/services/currency.service.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class CurrencyService {
 
   @Cacheable()
   getExchangeRate(fromCurrency, toCurrency, dt = new Date(), txnId?) {
-    const txnDt = moment(dt).format('y-MM-D');
+    const txnDt = dayjs(dt).format('YYYY-MM-D');
```
Comment (Julias0) [inline src/app/core/services/currency.service.ts:19]: `y` in moment is the short form for `YYYY`
This doesn't work in day js

File: src/app/core/interceptors/httpInterceptor.ts [hunk 2]
```diff
@@ -55,8 +55,8 @@ export class HttpConfigInterceptor implements HttpInterceptor {
 
   expiringSoon(accessToken: string): boolean {
     try {
-      const expiryDate = moment(this.jwtHelperService.getExpirationDate(accessToken));
-      const now = moment(new Date());
+      const expiryDate = dayjs(this.jwtHelperService.getExpirationDate(accessToken));
+      const now = dayjs(new Date());
       const differenceSeconds = expiryDate.diff(now, 'second');
```
Comment (rvab) [inline src/app/core/interceptors/httpInterceptor.ts:60]: Does `dayjs` too have the method? `.diff`?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -2434,11 +2434,11 @@ export class AddEditExpensePage implements OnInit {
...
     const today = new Date();
-    const minDate = moment(new Date('Jan 1, 2001'));
-    const maxDate = moment(new Date(today)).add(1, 'day');
-    const passedInDate = control.value && moment(new Date(control.value));
+    const minDate = dayjs(new Date('Jan 1, 2001'));
+    const maxDate = dayjs(new Date(today)).add(1, 'day');
+    const passedInDate = control.value && dayjs(new Date(control.value));
     if (passedInDate) {
-      return passedInDate.isBetween(minDate, maxDate)
+      return passedInDate.isBefore(maxDate) && passedInDate.isAfter(minDate)
```
Comment (rvab) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2441]: `dayjs`too has `isBetween`(https://day.js.org/docs/en/query/is-between) method. Does this not work?

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 4]
```diff
@@ -849,11 +849,11 @@ export class AddEditMileagePage implements OnInit {
...
     const today = new Date();
-    const minDate = moment(new Date('Jan 1, 2001'));
-    const maxDate = moment(new Date(today)).add(1, 'day');
-    const passedInDate = control.value && moment(new Date(control.value));
+    const minDate = dayjs(new Date('Jan 1, 2001'));
+    const maxDate = dayjs(new Date(today)).add(1, 'day');
+    const passedInDate = control.value && dayjs(new Date(control.value));
     if (passedInDate) {
-      return passedInDate.isBetween(minDate, maxDate)
+      return passedInDate.isBefore(maxDate) && passedInDate.isAfter(minDate)
```
Comment (rvab) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:856]: same as above

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { PopoverController } from '@ionic/angular';
-import * as moment from 'moment';
+import * as dayjs from 'dayjs';
```
Comment (rvab) [inline src/app/fyle/my-create-report/my-create-report.page.ts:4]: I don't see any usage.

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 6]
```diff
@@ -545,11 +545,11 @@ export class SplitExpensePage implements OnInit {
...
     const today = new Date();
-    const minDate = moment(new Date('Jan 1, 2001'));
-    const maxDate = moment(new Date(today)).add(1, 'day');
-    const passedInDate = control.value && moment(new Date(control.value));
+    const minDate = dayjs(new Date('Jan 1, 2001'));
+    const maxDate = dayjs(new Date(today)).add(1, 'day');
+    const passedInDate = control.value && dayjs(new Date(control.value));
     if (passedInDate) {
-      return passedInDate.isBetween(minDate, maxDate)
+      return passedInDate.isBefore(maxDate) && passedInDate.isAfter(minDate)
```
Comment (rvab) [inline src/app/fyle/split-expense/split-expense.page.ts:552]: same as above

File: src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ import { MatSnackBar } from '@angular/material/snack-bar';
 import { SnackbarPropertiesService } from '../../../../core/services/snackbar-properties.service';
 import { ToastMessageComponent } from 'src/app/shared/components/toast-message/toast-message.component';
 import { DeleteButtonComponent } from './delete-button/delete-button-component';
-import * as moment from 'moment';
+import * as dayjs from 'dayjs';
```
Comment (rvab) [inline src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.ts:13]: Don't see any usage.

**Review by rvab — CHANGES_REQUESTED** [review]

File: src/app/core/interceptors/httpInterceptor.ts (hunk 2 above)
Comment (Julias0) [reply src/app/core/interceptors/httpInterceptor.ts:60]: Yes

Comment (Julias0) [reply src/app/core/interceptors/httpInterceptor.ts:60]: dayjs is typsafe, so unreferenced stuff would come up as compiler errors in the unit tests & builds

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (Julias0) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2441]: It needs an extra plugin, which I assumed would just contribute to bundlesize without any extra benefit

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 4 above)
Comment (Julias0) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:856]: same as above

**Review by rvab — APPROVED** [review]

**Pull request merged**